### ISSUE TITLE
Enumerator#next_values behavior does not match MRI or docs

### ISSUE THREAD

**Issue by ryannevell** [jruby/jruby#8199]
**Environment Information**
jruby 9.4.6.0 (3.1.4) 2024-02-20 576fab2c51 OpenJDK 64-Bit Server VM 11.0.15+10-LTS on 11.0.15+10-LTS +jit [arm64-darwin]
MacOS Sonoma 14.2.1, M3 Max

**Expected Behavior**
https://ruby-doc.org/3.1.4/Enumerator.html#method-i-next_values describes the behavior for `Enumerator#next_values` with this table:

```
## yield args       next_values      next
#  yield            []               nil
#  yield 1          [1]              1
#  yield 1, 2       [1, 2]           [1, 2]
#  yield nil        [nil]            nil
#  yield [1, 2]     [[1, 2]]         [1, 2]
```

CRuby/MRI behaves as above for all versions tested:
- ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.arm64e-darwin23]
- ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [arm64-darwin23]
- ruby 3.1.4p223 (2023-03-30 revision 957bb7cb81) [arm64-darwin23]
- ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin23]

**Actual Behavior**
JRuby's behavior differs for the last 2 rows:
```
## yield args       next_values
#  yield nil        []     (expecting [nil])
#  yield [1, 2]     [1, 2] (expecting [[1, 2]])
```

**Test Code**
```
require 'minitest/autorun'
describe Enumerable do
  describe 'next_values' do
    it 'yield nil' do
      o = Object.new
      def o.each
        yield nil
      end
      _(o.to_enum.next_values).must_equal [nil]
    end

    it 'yield [1, 2]' do
      o = Object.new
      def o.each
        yield [1, 2]
      end
      _(o.to_enum.next_values).must_equal [[1, 2]]
    end
  end
end
```
**Output**
```
jruby 9.4.6.0 (3.1.4) 2024-02-20 576fab2c51 OpenJDK 64-Bit Server VM 11.0.15+10-LTS on 11.0.15+10-LTS +jit [arm64-darwin]
Run options: --seed 58630

# Running:

FF

Finished in 0.007662s, 261.0441 runs/s, 261.0441 assertions/s.

  1) Failure:
Enumerable::next_values#test_0001_yield nil [test.rb:13]:
Expected: [nil]
  Actual: []

  2) Failure:
Enumerable::next_values#test_0002_yield [1, 2] [test.rb:21]:
Expected: [[1, 2]]
  Actual: [1, 2]

2 runs, 2 assertions, 2 failures, 0 errors, 0 skips
```

**Comment by ryannevell** [comment]
Seems like there is already an existing test-case for one of these and it is a known issue from 8cb9455e4af8c36f8d121731a7b3d35215aae541

**Comment by headius** [comment]
Good find... I'm guessing you ran into this in real code? 

Shouldn't be hard to fix now that we know it is important.

**Comment by ryannevell** [comment]
Yes. I actually have a fix and am unblocked locally. I am going to work with my employer's legal team to hopefully get permission to share it.

**Comment by ryannevell** [comment]
Got approval. Submitted PR #8201

**Comment by headius** [comment]
Thanks for the fix! We'll be pushing out 9.4.7.0 in the next week.
